storage-proofs-core/src/gadgets/insertion.rs

```diff
@@ -316,6 +316,7 @@ pub fn select<Scalar: PrimeField, CS: ConstraintSystem<Scalar>>(
 }
 
 /// Takes two allocated numbers (`a`, `b`) and returns `a` if the condition is true, and `b` otherwise.
+#[allow(clippy::multiple_bound_locations)]
```

Comment:
Instead of this Clippy exception, I'd just remove the whole

```rust
where
    CS: ConstraintSystem<Scalar>,
```

as it's redundant anyway.